src/tests/electron/main.test.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as Electron from 'electron';
+import { Application } from 'spectron';
+
+describe('Electron E2E', () => {
+    let app: Application;
+
+    beforeAll(() => {
+        const electronPath = `${(global as any).rootDir}/drop/electron/extension/bundle/main.bundle.js`;
+        app = new Application({
+            path: Electron as any,
+            args: [electronPath],
+        });
+
+        return app.start();
+    });
+
+    test('test that app opened & loaded site', () => {
```

Comment:
Jest nowadays has support for `async` test methods that contain `await` statements; you can avoid a lot of the `.then` chaining boilerplate and instead do something like this:

```ts
test('test name', async () => {
    expect(app.browserWindow.isVisible()).toBe(true);
    expect(await app.client.getWindowCount()).toBe(2);
    expect(await app.webContents.getTitle()).toBe('the title');
});
```

Reply:
This is a big improvement, thanks! Unfortunately, making the change revealed an issue regarding spectron's typings (or lack thereof). I had to disable a tslint rule to make this work--I added a comment explaining why. Let me know if you think there's an improvement I can make